File: deck/base/code/native/node/runtime/environment.ts
// Environment over the host process. The working directory and environment variables, presented as one namespace so
// the native module docks it as a <global:environment> and forwards uniformly, matching the rust, swift, and kotlin
// shims. A missing variable reads as the empty string, so the public API stays total.
const environment = {
  currentDirectory: (): string => process.cwd(),
  getVariable: (name: string): string => process.env[name] ?? '',
}

File: deck/seed/code/native/rust/runtime/environment.rs
mod environment {
    pub fn current_directory() -> String {
        std::env::current_dir().map(|p| p.to_string_lossy().to_string()).unwrap_or_default()
    }
    pub fn get_variable(name: String) -> String {
        std::env::var(&name).unwrap_or_default()
    }
}
